sda-commons-server-s3/src/main/java/org/sdase/commons/server/s3/S3Bundle.java

```diff
@@ -111,11 +102,90 @@ public void run(C configuration, Environment environment) throws Exception {
     }
   }
 
+  private S3Client newClient() {
+    return S3Client.builder()
+        .region(getRegion())
+        .endpointOverride(URI.create(s3Configuration.getEndpoint()))
+        .serviceConfiguration(
+            software.amazon.awssdk.services.s3.S3Configuration.builder()
+                .pathStyleAccessEnabled(true)
+                .build())
+        .overrideConfiguration(
+            ClientOverrideConfiguration.builder()
+                .executionInterceptors(List.of(createTracingExecutingInterceptor()))
+                .putAdvancedOption(SdkAdvancedClientOption.SIGNER, createSigner(s3Configuration))
+                .build())
+        .httpClient(UrlConnectionHttpClient.builder().build())
+        .credentialsProvider(
+            createCredentialsProvider(
+                s3Configuration.getAccessKey(), s3Configuration.getSecretKey()))
+        .build();
+  }
+
+  public S3Presigner newPresigner() {
+    return S3Presigner.builder()
+        .s3Client(getClient())
+        .region(getRegion())
+        .credentialsProvider(
+            createCredentialsProvider(
+                s3Configuration.getAccessKey(), s3Configuration.getSecretKey()))
+        .build();
+  }
+
+  /**
+   * @return the region to use for the S3 client, default is 'eu-central-1'
+   */
+  private Region getRegion() {
+    String region = s3Configuration.getRegion();
+    if (region != null && !region.isBlank()) {
+      return Region.of(region);
+    }
+
+    // default
+    return Region.EU_CENTRAL_1;
```

Comment:
Does it make sense to put the default value in `S3Configuration` similar to the default value of `signerOverride`

Reply:
Why not. That makes it more transparent. The current documentation of the property in `S3Configuration` says "can be left empty" which might even be misleading. I've changed the default from `us-west-2` (v1) to `eu-central-1`. Usually customers will set their region anyway, e.g. when they host their own S3 server.